frontend/src/App.tsx

```diff
@@ -339,6 +341,16 @@ export class App extends PureComponent<Props, State> {
     })
 
     this.handleSessionStateChanged(sessionState)
+
+    // send current URL to server
+    this.sendCurrentUrlToServer()
```

Comment:
Does this happens every time query params are set "manually" in the browser?

Reply:
Nope, this only happens when the Streamlit app first loads. Every time query params are set "manually" in the browser -> this method will not be invoked.